File: rules/src/main/scala/net/michalsitko/kloc/game/Color.scala
package net.michalsitko.kloc.game

/**
 * Created with IntelliJ IDEA.
 * User: michal
 * Date: 5/26/13
 * Time: 1:18 PM
 * To change this template use File | Settings | File Templates.
 */
sealed abstract class Color () {
  def opposite(): Color
}

object Color {
  val whiteStr = "w"
  val blackStr = "b"

  def fromString(input: String): Option[Color] = {
    input match {
      case Color.whiteStr => Some(White())
      case Color.blackStr => Some(Black())
      case _ => None
    }
  }
}

case class White() extends Color (){
  def opposite(): Color = return Black()
  override def toString(): String = {
    Color.whiteStr
  }
}

case class Black() extends Color (){
  def opposite(): Color = return White()
  override def toString(): String = {
    Color.blackStr
  }
}

File: web/public/javascripts/game-rules/color.js
define([], function(){

    var white = {}
    var black = {};

    white.enemy = function(){
        return black;
    };

    white.toString = function(){
        return "w";
    }

    black.enemy = function(){
        return white;
    };

    black.toString = function(){
        return "b";
    }

    var Color = Object.freeze({
        white: white,
        black: black,
        fromString: function(str){
            switch(str){
                case "w":
                    return white;
                case "b":
                    return black;
                default:
                    return undefined;
            }
        }
    });

    return Color;
});